ethcore/src/block.rs

```diff
@@ -424,31 +423,9 @@ impl<'x> OpenBlock<'x> {
 
 	/// Turn this into a `LockedBlock`.
 	pub fn close_and_lock(self) -> Result<LockedBlock, Error> {
-		let mut s = self;
-
-		s.engine.on_close_block(&mut s.block)?;
-		s.block.state.commit()?;
-
-		if s.block.header.transactions_root().is_zero() || s.block.header.transactions_root() == &KECCAK_NULL_RLP {
-			s.block.header.set_transactions_root(ordered_trie_root(s.block.transactions.iter().map(|e| e.rlp_bytes())));
-		}
-		if s.block.header.uncles_hash().is_zero() || s.block.header.uncles_hash() == &KECCAK_EMPTY_LIST_RLP {
-			let uncle_bytes = encode_list(&s.block.uncles);
-			s.block.header.set_uncles_hash(keccak(&uncle_bytes));
-		}
-		if s.block.header.receipts_root().is_zero() || s.block.header.receipts_root() == &KECCAK_NULL_RLP {
```

Comment:
as mentioned before, this checks could lead to invalid block being locked, so I removed them